### PR TITLE
News List Layout

### PR DESCRIPTION
**Pull request by wiseodd**
News list and news item layout and quasi-implementation using RecyclerView.

5-swapped-url-comment copy

### CODE REVIEW

File: app/src/main/java/io/pumpkinz/pumpkinreader/etc/DividerItemDecoration.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import android.content.res.TypedArray;
+import android.graphics.Canvas;
+import android.graphics.Rect;
+import android.graphics.drawable.Drawable;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+
+
+public class DividerItemDecoration extends RecyclerView.ItemDecoration {
```
Comment (timotiusnc) [inline app/src/main/java/io/pumpkinz/pumpkinreader/etc/DividerItemDecoration.java:28]: What is this file for? Is it not included in support library?

File: app/src/main/java/io/pumpkinz/pumpkinreader/data/NewsViewHolder.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.widget.TextView;
+
+import io.pumpkinz.pumpkinreader.R;
+
+
+public class NewsViewHolder extends RecyclerView.ViewHolder {
```
Comment (timotiusnc) [inline app/src/main/java/io/pumpkinz/pumpkinreader/data/NewsViewHolder.java:11]: This file seems unused. What is it for?

File: app/src/main/java/io/pumpkinz/pumpkinreader/etc/DividerItemDecoration.java (hunk 1 above)
Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/etc/DividerItemDecoration.java:28]: For one reason or another, Google decided to not include this file. So I add it here.
This one is a decoration for RecyclerView, which will give the list divider between each list item.

File: app/src/main/java/io/pumpkinz/pumpkinreader/data/NewsViewHolder.java (hunk 2 above)
Comment (timotiusnc) [inline app/src/main/java/io/pumpkinz/pumpkinreader/data/NewsViewHolder.java:11]: Oh sorry, it's used for getting the `news_title`. OK

Comment (wiseodd) [inline app/src/main/java/io/pumpkinz/pumpkinreader/data/NewsViewHolder.java:11]: Check out NewsAdapter.java

File: app/src/main/res/layout/fragment_news_list.xml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".NewsListFragment">
```
Comment (timotiusnc) [inline app/src/main/res/layout/fragment_news_list.xml:5]: What's the use of `tools:context` here? I tried to delete it and the layout still worked.

Comment (wiseodd) [inline app/src/main/res/layout/fragment_news_list.xml:5]: I don't really understand either. It's the default, though.

**Pull request merged**